analysers/analyser_osmosis_highway_area_access.py

```diff
@@ -51,24 +51,120 @@
   )
 """
 
+sql20 = """
+CREATE TEMP TABLE {barriertype} AS
+SELECT
+  barrier.id AS id,
+  barrier.tags AS tags,
+  ways.id as wid,
+  ST_AsText(barrier.geom) AS geom
+FROM
+  nodes AS barrier
+  JOIN way_nodes ON
+    way_nodes.node_id = barrier.id
+  JOIN highways AS ways ON
+    ways.id = way_nodes.way_id AND
+    NOT ways.is_construction AND
+    NOT ways.is_area
+  LEFT JOIN highway_ends ON
+    ways.id = highway_ends.id AND
+    barrier.id = highway_ends.nid
+WHERE
+  barrier.tags != ''::hstore AND
+  barrier.tags?'barrier' AND
+  barrier.tags->'barrier' = '{barriertype}' AND
+  /* Default of bollard is access=no / bicycle=foot=yes; default of bus_trap is motor_vehicle=no / psv=foot=bicycle=yes.
+  Hence, the below three lines should cover all cases as long as we don't test for any non-motor_vehicle or vehicles under psv */
+  (NOT barrier.tags?'access' OR barrier.tags->'access' = 'no') AND
+  (NOT barrier.tags?'vehicle' OR barrier.tags->'vehicle' = 'no') AND
+  (NOT barrier.tags?'motor_vehicle' OR barrier.tags->'motor_vehicle' = 'no') AND
+  highway_ends.nid IS NULL -- Ignore transitions between ways that may have different tags
+"""
+
+sql21 = """
+SELECT
+    barrier.wid,
+    barrier.id,
+    barrier.geom,
+    highway.tags->'{vehicle}'
+FROM
+    {barriertype} AS barrier
+    JOIN highways AS highway ON
+      highway.id = barrier.wid AND
+      highway.tags?'{vehicle}' AND
+      NOT highway.tags->'{vehicle}' IN ('no', 'use_sidepath', 'unknown')
```

Comment:
Ouput `tags?'{vehicle}'` in sql20 and avoid a join here.

Reply:
Thanks! I did implement it slightly different by exporting all way tags, so I can avoid having to put sql20 in the loop (or having to check for all tags one by one here), but indeed much better without the join